### ISSUE TITLE
Azure SQL Database Deployment: Azure Active Directory Authentication against Azure SQL

### ISSUE THREAD

**Issue by tinodo** [Microsoft/azure-pipelines-tasks#8789]
## Issue Description
I am adding an Azure SQL Database Deployment task to my release pipeline in Azure DevOps. The SQL Server and SQL Database are deployed through an ARM template.  
I want to use an inline script to execute some SQL tasks against the newly created database.  
The Azure SQL Server is configured to use **Azure Active Directory** for authentication, and the service principal for Azure DevOps is listed as the sa account. This appears to be unsupported in the current version. I **have** to provide a SQL Login and Password.

The obvious way would be to use SQL Authentication, but then the password would have to be generated by the ARM template, after which I will be unable to expose it to the pipeline.

Can we support Azure AD Authentication in the Azure SQL Database Deployment task?

**Comment by structre40** [comment]
I wanted to comment on this as well that it is something that we would like to be able to use. I did notice in [source](https://github.com/Microsoft/azure-pipelines-tasks/blob/master/Tasks/SqlAzureDacpacDeploymentV1/DeploySqlAzure.ps1) that it is available but maybe has not been published yet for Azure DevOps?

**Comment by structre40** [comment]
I wanted to reply that I have a workaround by using Azure Inline Powershell task similar to:
```
$query ='select col from table'
$con = "Data Source=$(DatabaseServer);Initial Catalog=$(DatabaseName);User ID=$(ADDatabaseUser);Password='$(ADDatabasePassword)';Connect Timeout=30;Encrypt=False;Authentication='Active Directory Password'"
Invoke-Sqlcmd -Query $query -ConnectionString $con

```
I then use variables in my Azure DevOps release to run what is needed to be run. In my case it was setting an AAD Group to have access to a database so it can be used by my Azure App Service as a Service Principal.

**Comment by Ajay-MS** [comment]
@structre40 

We have already added AAD support to Azure SQL Dacpac deployment task. 

PR  - https://github.com/Microsoft/azure-pipelines-tasks/pull/9051 

It will be deployed within next 2-3 weeks.

**Comment by Ajay-MS** [comment]
@trisapeace 

Updated SQL Azure Dacpac Task will have an input for authentication type. 
![image](https://user-images.githubusercontent.com/16772764/52107355-fc471480-261b-11e9-8d05-05ce8ee7b09a.png)

**Comment by trisapeace** [comment]
Thanks for the quick response @Ajay-MS . And this will allow us to authenticate using a user who is part of a AD admin group to deploy the Azure SQL database, correct?

**Comment by Ajay-MS** [comment]
@trisapeace
  
Yes its true.

**Comment by Garand-Gary** [comment]
What is the "Active Directory - Integrated" option used for? In what user context does the task execute when this is selected? 

I assumed that it would run as the SPN of whatever service connection I select in the "Azure Subscription" dropdown but that doesn't seem to be the case for me.

I have created a service connection in DevOps, and then in AAD I have added the generated SPN to the group which is set as my Azure SQL AD Admin.

Using PowerShell I can get the token of my Azure DevOps SPN and connect to my database, and for testing I have run an arbitrary query, but when I try to use this in my Azure SQL Database Deployment task I only get a SqlPackage exit code 1 error.

I am striving for as much of a password-free life in Azure as I can get, and some sort of integrated security for my SQL Server deployments would be ideal.
